applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -416,7 +419,7 @@ class EigensolverStrategy
         if (rModelPart.NumberOfMasterSlaveConstraints() != 0) {
             this->pGetBuilderAndSolver()->ApplyConstraints(pScheme, rModelPart, rStiffnessMatrix, b);
         }
-        ApplyDirichletConditions(rStiffnessMatrix,-1.0);
```

Comment:
Because of the reasons stated in the comments, this change would produce wrong results for a `BlockBuilderAndSolver`

Reply:
Setting the factor to 0 here (and leaving the factor for the mass matrix at 1) will set the spurious eigenvalues to 0, which can be filtered easily. Maybe this is a way to proceed?